presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -452,13 +462,63 @@ private static boolean shouldUseFileSplitsFromInputFormat(InputFormat<?, ?> inpu
         }
 
         HiveDirectoryContext hiveDirectoryContext = new HiveDirectoryContext(recursiveDirWalkerEnabled ? RECURSE : IGNORED, cacheable);
+
+        if (HiveSessionProperties.isListFilesDisabled(session) && partition.isPresent()) {
```

Comment:
- what about non-partitioned tables?
- what if a table is mixed with file paths in some partitions and some not?